src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceBase.cs

```diff
@@ -419,5 +446,69 @@ public static DataSourceObjectMetadata CreateFolderMetadata(DataSourceObjectMeta
                     throw new ArgumentException($"Unsupported data source type \"{dataSourceType}\"", nameof(dataSourceType));
             }
         }
+
+        public static List<DatabaseInfo> ConvertToDatabaseInfo(DataSourceType dataSourceType, List<DataSourceObjectMetadata> clusterDBDetails)
+        {
+            switch (dataSourceType)
```

Comment:
Do we even need this param. I don't see anything specific to Kusto being done here. Just a conversion from DataSourceObjectMetadata to DatabaseInfo.